### ISSUE TITLE
Help: KafkaClient in client_async.py fails in python3

### ISSUE THREAD

**Issue by sondinhinfectious** [dpkp/kafka-python#532]
I'm running python 3.4.2 with Kafka 0.8.1.1

I've had a quick look and submitted a (broken) PR.

Issue looks to be that ConnectionError isn't defined in PY3:

At line 23 if `kafka.client_async`, `six.PY2 == True` defines `ConnectionError = None` but six.PY3 will not assign it to anything. The assertion checking for the exception type at line 628 will then fail.

The follow snippet demonstrates this under 3.4.2:

```
>>> import six
>>> if six.PY2:
...     ConnectionError = None
... 
>>> f = IOError
>>> assert isinstance(f, ConnectionError)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

**Comment by dpkp** [comment]
ConnectionError is a valid python3 exception -- if it were not, you would get a NameError. Try `isinstance(f, Foobar)` for example. It is *not* a valid exception in python2, however, so that code block is included simply to avoid compilation errors when referencing ConnectionError from a python3-specific error handling block later.

What is the actual error you are seeing? Can you use ipython and run %pdb after the error ?

**Comment by sondinhinfectious** [comment]
Sorry for the late reply.

Here's what I got and shows why the `RequestTimedOutError` change in my PR worked for me (and just me); changing that got the `d56d2faddabd36ecbec73954bd4a8f17ed501e84` release working.

```
(python3.4.2) son.dinh@dev-1:~/kafka-python$ ipython
WARNING: Readline services not available or not loaded.
WARNING: The auto-indent feature requires the readline library
Python 3.4.4 (default, Feb  3 2016, 15:38:51) 
Type "copyright", "credits" or "license" for more information.

IPython 4.1.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from kafka import KafkaConsumer

In [2]: %pdb
Automatic pdb calling has been turned ON

In [3]: consumer = KafkaConsumer("impression", bootstrap_servers=["kafka-1", "kafka-2", "kafka-3"], group_id="kb_debug")
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-6a4a421e138d> in <module>()
----> 1 consumer = KafkaConsumer("impression", bootstrap_servers=["kafka-1", "kafka-2", "kafka-3"], group_id="kb_debug")

/home/son.dinh/kafka-python/kafka/consumer/group.py in __init__(self, *topics, **configs)
    180         # Check Broker Version if not set explicitly
    181         if self.config['api_version'] == 'auto':
--> 182             self.config['api_version'] = self._client.check_version()
    183         assert self.config['api_version'] in ('0.9', '0.8.2', '0.8.1', '0.8.0'), 'Unrecognized api version'
    184 

/home/son.dinh/kafka-python/kafka/client_async.py in check_version(self, node_id)
    626                 assert f.exception.args[0].errno in (32, 54, 104)
    627             else:
--> 628                 assert isinstance(f.exception.args[0], ConnectionError)
    629             log.info("Broker is not v%s -- it did not recognize %s",
    630                      version, request.__class__.__name__)

AssertionError: 
> /home/son.dinh/kafka-python/kafka/client_async.py(628)check_version()
    626                 assert f.exception.args[0].errno in (32, 54, 104)
    627             else:
--> 628                 assert isinstance(f.exception.args[0], ConnectionError)
    629             log.info("Broker is not v%s -- it did not recognize %s",
    630                      version, request.__class__.__name__)

ipdb> f
<kafka.future.Future object at 0x7f008ddf4940>
ipdb> f.exception
RequestTimedOutError('Request timed out after 40000 ms',)
ipdb> f.exception.args
('Request timed out after 40000 ms',)
```

**Comment by dpkp** [comment]
I've updated the check_version code to be a bit more forgiving. Fix is in master @ 4d17ec2
